### PR TITLE
feat: introduce separate Bool kind

### PR DESCRIPTION
**Pull request by mlutze**
depends on #5943. Followups are: use it for Choice type, fix type formatting

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Conflicted and probably needs to be rebased :/

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala [hunk 1]
```diff
@@ -1457,7 +1457,15 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
...
+      Or ~ zeroOrMore(WS ~ atomic("&") ~ WS ~ Type ~ SP ~> ParsedAst.Type.Intersection)
+    }
+
+    def Or: Rule1[ParsedAst.Type] = rule {
+      And ~ zeroOrMore(WS ~ atomic("or") ~ WS ~ Type ~ SP ~> ParsedAst.Type.Or)
+    }
+
+    def And: Rule1[ParsedAst.Type] = rule {
+      Ascribe ~ zeroOrMore(WS ~ atomic("and") ~ WS ~ Type ~ SP ~> ParsedAst.Type.And)
     }
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1470]: How is not parsed?

File: main/src/ca/uwaterloo/flix/language/phase/MonoTyper.scala [hunk 2]
```diff
@@ -224,18 +224,17 @@ object MonoTyper {
 
             case TypeConstructor.Record => args.head
 
+            case TypeConstructor.False => MonoType.Unit
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/MonoTyper.scala:227]: You mostly have True first, but sometimes False is first :)

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala (hunk 1 above)
Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1470]: Good question. It's not.

The parser is likely going to need adjustment anyway.

**Comment by mlutze** [comment]
There's more to be done here since there's no Boolean unification enabled for this Boolean kind. I'm looking at how best to parameterize our current stuff over different kinds.

**Pull request merged**